jupyterlab_gcpscheduler/src/components/action_bar.tsx

```diff
@@ -39,14 +44,52 @@ const actionBar = style({
   ...csstips.endJustified,
 });
 
+const actionBarContainer = style({
+  paddingTop: '16px',
+});
+
+const actionBarDisplayMessage = style({
+  paddingTop: '9px',
+  ...csstips.horizontal,
+  color: COLORS.caption,
+  fontSize: '12px',
+});
+
 /** Funtional Component for defining an action bar with buttons. */
-export function ActionBar(props: Props) {
-  return (
-    <div className={actionBar}>
-      <Button onClick={props.onDialogClose}>
-        {props.closeLabel || 'Close'}
-      </Button>
-      {props.children}
-    </div>
-  );
+export class ActionBar extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { displayMessage: props.displayMessage };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.displayMessage !== this.props.displayMessage) {
```

Comment:
What is the advantage to having `displayMessage` stored in state since it looks like it will always reflect what was provided with by props and is never changed by anything internal to the component?

Reply:
It needs to change when the cron string is changed in this case

Reply:
But since `ActionBar` doesn't modify its own state, wouldn't it just received the updated props from its parent?

Reply:
I guess ideally, but it wasn't re-rendering without setting the state explicitly

Reply:
Interesting. I checked this out locally and switched back to the functional implementation and it seemed to work as intended.

```
/** Funtional Component for defining an action bar with buttons. */
export function ActionBar(props: Props) {
  return (
    <Grid container spacing={1} className={STYLES.actionBarContainer}>
      {props.displayMessage && (
        <Grid item sm={8}>
          <span className={STYLES.actionBarDisplayMessage}>
            {props.displayMessage}
          </span>
        </Grid>
      )}
      <Grid item sm={props.displayMessage ? 4 : 12}>
        <div className={STYLES.actionBar}>
          {props.closeOnRight && props.children}
          <Button onClick={props.onDialogClose}>
            {props.closeLabel || 'Close'}
          </Button>
          {!props.closeOnRight && props.children}
        </div>
      </Grid>
    </Grid>
  );
}
```

Reply:
Oh wow, okay I can try it out again, maybe I did something wrong when testing it.

Reply:
Fixed! Thanks for looking into it! I have no idea why it didn't work the first time.